src/vs/workbench/contrib/notebook/browser/notebookOptions.ts

```diff
@@ -547,34 +548,37 @@ export class NotebookOptions extends Disposable {
 		return this._layoutConfiguration.cellStatusBarHeight;
 	}
 
-	private _computeBottomToolbarDimensions(compactView: boolean, insertToolbarPosition: 'betweenCells' | 'notebookToolbar' | 'both' | 'hidden', insertToolbarAlignment: 'left' | 'center', cellToolbar: 'right' | 'left' | 'hidden'): { bottomToolbarGap: number; bottomToolbarHeight: number } {
+	private _computeBottomToolbarDimensions(compactView: boolean, insertToolbarBetweenCells: boolean, insertToolbarAlignment: 'left' | 'center', cellToolbar: 'right' | 'left' | 'hidden'): { bottomToolbarGap: number; bottomToolbarHeight: number } {
+		if (!insertToolbarBetweenCells) {
+			return compactView ? {
+				bottomToolbarGap: 0,
+				bottomToolbarHeight: 0
+			} : {
+				bottomToolbarGap: 12,
```

Comment:
this is enough gap for the `cellToolbar` to have a gap between the cell and the previous cell (in the case the previous cell has no output)